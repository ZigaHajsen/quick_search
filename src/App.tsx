import { useState } from 'react';
import Search from './components/Search';
import Result from './components/Result';

import states from './state_capitals.json';

const App = () => {
  const [search, setSearch] = useState('');

  let matches = states.filter((state) => {
    const regex = new RegExp(`^${search}`, 'gi');
    return state.name.match(regex) || state.abbr.match(regex);
  });

  if (search.length === 0) {
    matches = [];
  }

  return (
    <div>
      <Search setSearch={setSearch} />
      <Result matches={matches} />
    </div>
  );
};

export default App;
