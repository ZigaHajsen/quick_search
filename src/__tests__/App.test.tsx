import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import App from '../App';

test('<App /> matches snapshot', async () => {
  const { getByPlaceholderText, getByText, container } = render(<App />);

  expect(container).toMatchSnapshot();

  const input = getByPlaceholderText('Search Capital');

  fireEvent.change(input, { target: { value: 'ma' } });

  getByText('Maine (ME) Augusta');
  getByText('Maryland (MD) Annapolis');
  getByText('Massachusetts (MA) Boston');
});
