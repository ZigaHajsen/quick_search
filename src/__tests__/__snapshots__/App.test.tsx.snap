// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div>
  <div>
    <div>
      <h3>
        State Capital Lookup
      </h3>
      <input
        name="search"
        placeholder="Search Capital"
        type="text"
      />
    </div>
    <div />
  </div>
</div>
`;
